### PR TITLE
Add enterprise commit to buildinfo

### PR DESCRIPTION
**Pull request by kminehart**
Analogous to https://github.com/grafana/grafana/pull/75242

### CODE REVIEW

File: containers/buildinfo.go [hunk 1]
```diff
@@ -40,6 +47,8 @@ func GetBuildInfo(ctx context.Context, d *dagger.Client, dir *dagger.Directory,
 		WithMountedDirectory("/src", dir).
 		WithWorkdir("/src")
 
+	enterpriseSha, _ := enterpriseCommit(ctx, container)
```
Comment (evictorero) [inline containers/buildinfo.go:50]: This is the only use of this func, if we don't care about the `err` we could remove it from the func.

File: containers/buildinfo.go [hunk 2]
```diff
@@ -57,15 +66,37 @@ func GetBuildInfo(ctx context.Context, d *dagger.Client, dir *dagger.Directory,
...
 	return result, nil
 }
 
+func enterpriseCommit(ctx context.Context, container *dagger.Container) (string, error) {
+	var err error
+	c := container.
+		WithEntrypoint([]string{}).
+		WithExec([]string{"/bin/sh", "-c", "cat /src/.enterprise-commit || return 0"})
+
+	log.Println("Getting container exit error")
```
Comment (evictorero) [inline containers/buildinfo.go:85]: I think we forgot to remove this from debugging.

File: containers/buildinfo.go [hunk 3]
```diff
@@ -57,15 +66,37 @@ func GetBuildInfo(ctx context.Context, d *dagger.Client, dir *dagger.Directory,
...
+	c, err = ExitError(ctx, c)
+	if err != nil {
+		return "", nil
+	}
+
+	log.Println("Getting container stdout")
```
Comment (evictorero) [inline containers/buildinfo.go:91]: Debugging?

File: containers/buildinfo.go [hunk 4]
```diff
@@ -57,15 +66,37 @@ func GetBuildInfo(ctx context.Context, d *dagger.Client, dir *dagger.Directory,
...
-		Commit:    sha,
-		Branch:    branch,
-		Timestamp: timestamp,
+		Version:          version,
+		Commit:           sha,
+		EnterpriseCommit: enterpriseSha,
+		Branch:           branch,
+		Timestamp:        timestamp,
 	}
 	span.SetAttributes(attribute.String("version", version), attribute.String("commit", sha), attribute.String("branch", branch), attribute.String("timestamp", timestamp.Format(time.RFC3339)))
```
Comment (evictorero) [inline containers/buildinfo.go:75]: I'm not sure about the goal of this `span.SetAttributes` but shouldn't we add the `enterpriseCommit` here when not empty?

**Review by evictorero — COMMENTED** [review]
Leave some nit comments but looks good. Thanks for your help!

File: containers/buildinfo.go (hunk 4 above)
Comment (kminehart) [reply containers/buildinfo.go:75]: That's a good idea, this is for tracing / telemetry.

File: containers/buildinfo.go (hunk 1 above)
Comment (kminehart) [reply containers/buildinfo.go:50]: Yeah good call.

File: containers/enterprise.go [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
+		WithEntrypoint([]string{}).
+		WithExec([]string{"/bin/sh", "-c", "git rev-parse --short HEAD > .enterprise-commit"}).
+		File("/src/grafana-enterprise/.enterprise-commit")
+
+	// Initializes Grafana Enterprise in the Grafana directory
+	g := d.Container().From(BusyboxImage).
 		WithDirectory("/src/grafana", grafana).
 		WithDirectory("/src/grafana-enterprise", enterprise).
 		WithWorkdir("/src/grafana-enterprise").
+		WithFile("/src/grafana/.enterprise-commit", hash).
```
Comment (zerok) [inline containers/enterprise.go:22]: That file should probably be deleted before packaging, right?

**Review by zerok — APPROVED** [review]

File: containers/enterprise.go (hunk 5 above)
Comment (kminehart) [reply containers/enterprise.go:22]: packaging uses an explicit list of paths to package so it won't be included.

File: containers/buildinfo.go (hunk 4 above)
Comment (zerok) [reply containers/buildinfo.go]: Done :)

**Pull request merged**